e2e/_suites/kubernetes-autodiscover/features/heartbeat.feature

```diff
@@ -0,0 +1,25 @@
+@kubernetes
+@autodiscover
+@beats
+Feature: Monitor uptime of Kubernetes resources
+  Use Kubernetes autodiscover features in Heartbeat to monitor pods, services and nodes.
+
+Scenario: Monitor pod availability using hints with named ports
+  Given "heartbeat" is running with "hints enabled for pods"
+   When "redis" is deployed with "monitor annotations with named port"
+   Then "heartbeat" collects events with "kubernetes.pod.name:redis"
+    And "heartbeat" collects events with "monitor.status:up"
+
+Scenario: Monitor service availability using hints
+  Given "heartbeat" is running with "hints enabled for services"
+   When "redis service" is deployed with "monitor annotations"
+   Then "heartbeat" collects events with "kubernetes.service.name:redis"
+    And "heartbeat" collects events with "monitor.status:down"
+   When "redis" is running
```

Comment:
Is it possible to split this into two separate scenarios? It's not common to have a `When` after a `Then`. It makes readers to think in a scripted (complex) scenario rather than in a straight forward one.

For me this is the only blocker. Other than that, the PR is GOLD 🥇